javascript/src/URDFClasses.js

```diff
@@ -242,9 +246,81 @@ class URDFJoint extends URDFBase {
             }
 
             case 'floating':
+                // no-op if all values are identical to existing value or are null
+                if (this.jointValue.every((value, index) => values[index] === value || values[index] === null)) return didUpdate;
+                // anonymous block scope to prevent variable name clobbering with other cases in the switch
+                {
```

Comment:
Nit: can we follow the code formatting in the rest of the and wrap the whole case in braces:

```js
case 'floating': {

  // ...

}
```

Reply:
yep. worth noting this is not just styling: this is creating an anonymous block scope.